src/ngraph/op/experimental/strided_slice.hpp

```diff
@@ -0,0 +1,120 @@
+//*****************************************************************************
```

Comment:
I think of StridedSlice as v1 of Slice (https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/slice.hpp)
We could leave out DynSlice unversioned and eventually drop it from experimental. Upgrade/Downgrade should convert ops between v0::Slice and v1::StridedSlice

Reply:
I see the the following disadvantages of introduction `StridedSlice` as v1 of `Slice`:
- during downgrade pass: `begin`, `end`, `strides` must be constant
- during downgrade pass: conversion `begin_mask` and `end_mask` seems to be easy, but handling `new_axis_mask`, `shrink_axis_mask`, `ellipsis_mask` is rather problematic
The only limit of using  `StridedSlice` as v1 of `DynSlice` which I see is requirement of static rank of input during upgrade pass.
Execution downgrade pass for `dynamic backend` was added anyway for `Reshape:v1`.
In case of op location I think you are right that `experimental` is probably not good place for `StridedSlice:v1` (for `DynSlice` I don't know?)
@jbobba  What do you think about these arguments? Can we leave `StrideSlice` as v1 of `DynSlice`? :)

Reply:
During downgrade passes, we expect certain inputs to be constants anyway. For dealing with new_axis_mask, shrink_mask etc. isn't it the same as https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/pass/dyn_elimination.cpp#L180

Reply:
@jbobba Using `Slice` as `v0` as `StridedSlice` was implemented